src/components/needle.js

```diff
@@ -1,12 +1,13 @@
-import React from 'react';
-import d3Shape from "d3-shape";
+import React from "react";
 
 export default class Needle extends React.Component {
+
   render() {
+    const {rotation, height} = this.props;
     return (
-      <g transform={`rotate(50)`}>
+      <g transform={`rotate(${rotation})`}>
         <path
-          d="M 0 5 C -1,5 -4,3 -6,0 L 0 -175 L 6 0 C 4,3 1,5 0,5"
+          d={`M 0 5 C -1,5 -4,3 -6,0 L 0 -${height} L 6 0 C 4,3 1,5 0,5`}
           stroke="black"
```

Comment:
Understood, I've moved out `stroke` and `fill`, etc. into the `defaultStyles` object. I couldn't move `d` attribute values too, because they're dependent upon the radius of the gauge, and they don't work within the `style` props object, so that has been moved into the `renderNeedle` function, but I reckon this should be overridable somehow, correct? If so how should the user input their own `d` values if they desire?